text/0000-update_asset_packaging.md

```diff
@@ -0,0 +1,109 @@
+- Feature Name: update_asset_packaging
+- Start Date: 2016-05-10
+- RFC PR:
+- Issue:
+
+# Summary
+[summary]: #summary
+
+Several suggestions have been made to improve the way we handle our assets packaging.
+This RFC attempts to present the various options so that a decision can be made.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, most assets (js/css) are used as gems and managed by bundler.
+A few are included directly in the vendor/ directory due to lack of gems.
+Some of the reasons that we may want to change the way we currently handle this:
+* Some assets are not packaged as gems, so we need to either vendor or package them ourselves.
+* Some asset gems are not updated at the same rate as the upstream source.
+* Each gem dependency must be packaged separately as RPM/DEB.
+* Difficult to integrate newer JS technology such as es2015 or js modules.
+* Assets included in the vendor/ directory need manual updating when new versions are released.
+
+
+# Detailed design
+[design]: #detailed-design
+
+NPM will be used to manage as many of the assets that are currently being vendored or bundled as possible.
+If we run into issues with packages that are only available for Bower in the future, we will consider adding Bower as well. The main advantage of using only NPM is that we have a central location for all js dependencies and only one tool added rather then two, which reduces the complexity. NPM also allows namespacing JS modules using CommonJS to prevent possible collisions between various modules.
+The node modules will be integrated into the asset pipeline using one of the methods detailed under the Alternatives heading below.
+As we will currently continue using the Rails assets pipeline, any plugin will be able to use whatever method they choose to handle any assets as long as the result is added to the pipeline (as is done currently).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Some reasons to stay with the current solution:
+* No extra work needed.
+* Single package management system - bundler. All dependencies are in one central location, build process is simple.
+* Rails asset pipeline doesn’t require any modifications to handle the current solution.
+
+# Alternatives
+[alternatives]: #alternatives
+
+__Option 1: Use Grunt task runner__
+
+_Pros:_
+* Existing knowledge base and tasks from Katello/Bastion.
+
+_Cons:_
+* Each task needs manual set up and configuration.
+* Creates multiple intermediate files while running.
+
+__Option 2: Use Gulp task runner__
+
+_Pros:_
+* Runs faster then Grunt and with less configuration.
+* Tasks can be built as streams of smaller units.
+
+_Cons:_
+* Each task needs manual set up and configuration.
+* Little relevant knowledge within the current core team.
+
```

Comment:
I'm not sure why we're adding grunt and gulp are in here since you can use webpack with [grunt](http://webpack.github.io/docs/usage-with-grunt.html) and [gulp](http://webpack.github.io/docs/usage-with-gulp.html).  To me, these accomplish separate things.